File: src/main/scala/com/barley/model/Attachment.scala
package com.barley.model

import java.lang
import java.util.Date
import javax.persistence.{Column, Entity, GeneratedValue, Id}

import scala.beans.BeanProperty

/**
  * Created by vikram.gulia on 11/24/15.
  */
@Entity
class Attachment {

  @Id
  @GeneratedValue
  @BeanProperty
  var attachment_id: lang.Long = _
  @BeanProperty
  @Column(name = "message_id")
  var messageId: lang.Long = _
  @BeanProperty
  var attachment_type: String = _
  @BeanProperty
  var file_id: lang.Long = _
  @BeanProperty
  var date_created: Date = _

}

File: src/main/java/com/barley/model/Attachment.java
package com.barley.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Attachment {
    @Id
    @GeneratedValue
    private Long attachment_id;
    @Column(name = "message_id")
    private Long messageId;
    private String attachment_type;
    private Long file_id;
    private Date date_created;

    public Long getAttachment_id() {
        return attachment_id;
    }

    public void setAttachment_id(Long attachment_id) {
        this.attachment_id = attachment_id;
    }

    public Long getMessageId() {
        return messageId;
    }

    public void setMessageId(Long messageId) {
        this.messageId = messageId;
    }

    public String getAttachment_type() {
        return attachment_type;
    }

    public void setAttachment_type(String attachment_type) {
        this.attachment_type = attachment_type;
    }

    public Long getFile_id() {
        return file_id;
    }

    public void setFile_id(Long file_id) {
        this.file_id = file_id;
    }

    public Date getDate_created() {
        return date_created;
    }

    public void setDate_created(Date date_created) {
        this.date_created = date_created;
    }
}
